cmake/tribits/common_tpls/FindTPLNetcdf.cmake

```diff
@@ -43,7 +43,7 @@ endif()
 
 set(Netcdf_ALLOW_MODERN FALSE CACHE BOOL "Allow finding Netcdf as a modern CMake config file with exported targets (and only this way)")
 
-if (Netcdf_ALLOW_MODERN  AND  TARGET HDF5::HDF5)
+if (Netcdf_ALLOW_MODERN)
```

Comment:
The danger of not checking to see if the modern HDF5 was found is that you can end up with a situation where the modern find_package(HDF5 CONFIG ...) commands fails and this find_package(netCDF CONFIG ...) command runs which will result in a configure-time error because it is missing the necessary HDF5 targets.

We need some type of check that the correct HDF5 module was found.  How can we do that?

Reply:
Sure, I agree.  Let's define a var called something like `HDF5_FOUND_MODERN_CONFIG_FILE=TRUE`.